lib/Common/Memory/RecyclerTelemetryInfo.cpp

```diff
@@ -0,0 +1,256 @@
+            if (this->hostInterface != nullptr)
+            {
+                LPFILETIME ft = this->hostInterface->GetLastScriptExecutionEndTime();
+                this->lastPassStats->lastScriptExecutionEndTime = *ft;
+            }
+
+            this->lastPassStats->processCommittedBytes_start = RecyclerTelemetryInfo::GetProcessCommittedBytes();
+            this->lastPassStats->processAllocaterUsedBytes_start = PageAllocator::GetProcessUsedBytes();
+            this->lastPassStats->isInScript = this->recycler->GetIsInScript();
+            this->lastPassStats->isScriptActive = this->recycler->GetIsScriptActive();
+
+            this->FillInSizeData(this->recycler->GetHeapInfo()->GetRecyclerLeafPageAllocator(), &this->lastPassStats->threadPageAllocator_start);
+            this->FillInSizeData(this->recycler->GetHeapInfo()->GetRecyclerPageAllocator(), &this->lastPassStats->recyclerLeafPageAllocator_start);
+            this->FillInSizeData(this->recycler->GetHeapInfo()->GetRecyclerLargeBlockPageAllocator(), &this->lastPassStats->recyclerLargeBlockPageAllocator_start);
+#ifdef RECYCLER_WRITE_BARRIER_ALLOC_SEPARATE_PAGE
+            this->FillInSizeData(this->recycler->GetHeapInfo()->GetRecyclerWithBarrierPageAllocator(), &this->lastPassStats->recyclerWithBarrierPageAllocator_start);
+#endif
+            this->lastPassStats->startPassProcessingElapsedTime = Js::Tick::Now() - start;
+        }
+
+
+    }
+
+
+    void RecyclerTelemetryInfo::EndPass()
+    {
+        if (!this->captureTelemetryData)
+        {
+            return;
+        }
+
+        Js::Tick start = Js::Tick::Now();
+
+        AssertOnMainThread(this, RecyclerTelemetryInfo::EndPass);
+        //Output::Print(L"EndPass called (current thread = %p, main thread = %p), passCount  = %i\r\n", GetCurrentThreadContextId(), mainThreadID, this->passCount);
+
+        this->lastPassStats->isGCPassActive = false;
+        this->lastPassStats->passEndTimeTick = Js::Tick::Now();
+
+        this->lastPassStats->processCommittedBytes_end = RecyclerTelemetryInfo::GetProcessCommittedBytes();
+        this->lastPassStats->processAllocaterUsedBytes_end = PageAllocator::GetProcessUsedBytes();
+
+        this->FillInSizeData(this->recycler->GetHeapInfo()->GetRecyclerLeafPageAllocator(), &this->lastPassStats->threadPageAllocator_end);
+        this->FillInSizeData(this->recycler->GetHeapInfo()->GetRecyclerPageAllocator(), &this->lastPassStats->recyclerLeafPageAllocator_end);
+        this->FillInSizeData(this->recycler->GetHeapInfo()->GetRecyclerLargeBlockPageAllocator(), &this->lastPassStats->recyclerLargeBlockPageAllocator_end);
+#ifdef RECYCLER_WRITE_BARRIER_ALLOC_SEPARATE_PAGE
+        this->FillInSizeData(this->recycler->GetHeapInfo()->GetRecyclerWithBarrierPageAllocator(), &this->lastPassStats->recyclerWithBarrierPageAllocator_end);
+#endif
+
+        // get bucket stats
+        Js::Tick bucketStatsStart = Js::Tick::Now();
+        BucketStatsReporter bucketReporter(this->recycler);
```

Comment:
This is mentioned in the TODO already to see if this has benchmarks impact.  Also be on the look out for perflab CPU time regressions.   An extra pass going thru all the heap block on each GC is probably going to show up.